rclpy/rclpy/__init__.py

```diff
@@ -109,6 +110,7 @@ def spin_once(node, *, timeout_sec=None):
         if request:
             response = srv.callback(request, srv.srv_type.Response())
             srv.send_response(response, header)
+    gc.collect()
```

Comment:
Yeah, sorry if I miscommunicated, but we don't want to do this.

From the other thread I was just saying that we should definitely have the destructor call the fini function in C, but we should let the gc run when it feels like it. Even though that might throw valgrind off.

The rest of the patch can stay though I think (still haven't reviewed it yet).